pystac_client/warnings.py

```diff
@@ -0,0 +1,73 @@
+from contextlib import contextmanager
+import warnings
+
+
+FALLBACK_MSG = "Falling back to pystac. This might be slow."
+
+
+class PystacClientWarning(UserWarning):
+    """Base warning class"""
+
+    ...
+
+
+class NoConformsTo(PystacClientWarning):
+    """Inform user when client does not have "conformsTo" set"""
+
+    ...
+
+
+class DoesNotConformTo(PystacClientWarning):
+    """Inform user when client does not conform to extension"""
+
+    ...
+
+
+class MissingLink(PystacClientWarning):
+    """Inform user when link is properly implemented"""
+
+    ...
+
+
+class FallbackToPystac(PystacClientWarning):
+    """Inform user when falling back to pystac implementation"""
+
+    ...
+
+
+@contextmanager
+def strict():  # type: ignore
+    """Context manager for raising all pystac-client warnings as errors
+
+    For more fine-grained control or to set filter warnings in the whole
+    python session, use the ``warnings`` module directly.
+
+    Examples:
+
+    >>> import warnings
+    ... warnings.filterwarnings("error", category=FallbackToPystac)
```

Comment:
```suggestion
    >>> import warnings
    >>> # Turn all `FallbackToPystac` warnings into errors
    >>> warnings.filterwarnings("error", category=FallbackToPystac)
```

It's been a while since I've looked at interpreter-in-docstring stuff, but I thought `...` was continuation, and `>>>` was for a new statement? I could very well be wrong...

Also, added a comment explaining what we're doing in the example.

Reply:
No I think you are right.